wiring/inc/spark_wiring_system.h

```diff
@@ -148,8 +148,18 @@ class SystemClass {
 
     inline int enableFeature(HAL_Feature feature)
     {
+        /* PRE-ENABLE TASKS */
+        bool init_sram = false;
+        // Initialize SRAM on first use of retained memory
+        if (feature==FEATURE_RETAINED_MEMORY && !HAL_Feature_Get(FEATURE_RETAINED_MEMORY)) {
+            init_sram = true;
+        }
+
+        /* ENABLE FEATURE */
         int result = HAL_Feature_Set(feature, true);
-        if (feature==FEATURE_RETAINED_MEMORY && !HAL_Feature_Get(FEATURE_WARM_START)) {
+
+        /* POST-ENABLE TASKS */
+        if (feature==FEATURE_RETAINED_MEMORY && init_sram) {
```

Comment:
Ok, good thing I went and tested this... because I found a new problem.  Before that though, initializing one `retained int` takes less than 1us (the GPIO toggling is taking some time... not sure how much, let's say very little).  Conversely initializing a `retained uint8_t array[4090] = {0};` took about 220us.  So not bad, and it's dependent on how much you are using which is great.

The problem I found, was that if I leave VBAT floating when powering up for the first time, SRAM remains uninitialized.  This was not expected as the first time we are powering up, the Backup Regulator Ready bit should be cleared.  If I attached VBAT to GND with a wire, it powers up that first time initialized.  If I attach VBAT to 3V3 with a wire, it typically also powers up initialized.  Only if floating do I typically see it not initialized.  This means when it's floating, sometimes the BRR bit is being read as SET on power up... which is bizarre.

If you are using this feature for Backup RAM, it is recommended then to have VBAT connected to a known good power source on system boot.

If you are using this feature for Extra RAM and want to ensure it always initializes, it is recommended to jumper VBAT to GND.